learn/react-native/pdf-preview-and-download.md

```diff
@@ -0,0 +1,73 @@
+---
+title: "PDF preview"
+id: "pdf-preview-and-download"
+sidebar_label: "PDF preview"
+---
+---
+import PdfPreview from '/learn/assets/pdf-preview-and-download.png';
+
+Viewing PDFs without leaving the mobile app offers ability for developers to keep their users in the app's context. It is possible to let PDF file download to the phone and have the user open it using a different app. That may not be best user experience for an app. Not all users may know how to locate and open the PDF file in a different app. 
+In this blog we will see how to add PDF preview functionality to a WaveMaker app using the `react-native-pdf` and 
+`react-native-blob-util` libraries.
+
+## Adding PDF Preview in a WaveMaker App
+
+The `react-native-pdf` library is a React Native PDF viewer component with support for both iOS and Android. 
+It uses the native libraries `PDFKit` on iOS and `PdfRenderer` on Android. 
+The `react-native-blob-util` library is a React Native wrapper for the `blob-util` library.
+We can leverage these libraries to add PDF preview functionality to a WaveMaker app. The best way to accomplish this is 
+through creating a prefab that will handle the PDF functional component.
+Please refer to this [page](https://docs.wavemaker.com/learn/react-native/custom-widget) for more information on how to 
+create a prefab that can be used in WaveMaker React Native project.
+
+
+## Downloading PDFs in a WaveMaker App
+
+Downloading PDFs is a common requirement in mobile applications, and React Native makes it relatively straightforward to implement this functionality. 
```

Comment:
Does this work for PDFs which are streamed? for example, there is assumption here that all of the pdf content is avaialble in the viewer and blob content will give the file. Can the download action be bound to another REST API endpoint that triggers the same file download?

Reply:
@shilwantkumar can you please check this?

Reply:
We are checking whether the preview works for password protected PDFs or not.

Reply:
@prasanth-reddy-wavemaker-com we checked preview for  password protected pdf is working